controllers/backup.go

```diff
@@ -232,34 +233,67 @@ func (r *ClusterGroupUpgradeReconciler) backupActive(ctx context.Context, cluste
 
 }
 
-// setBackupRequired sets conditions of backup required
-func (r *ClusterGroupUpgradeReconciler) setBackupRequired(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
-
-	meta.SetStatusCondition(
-		&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
-			Type:    BackupStateDone,
-			Status:  metav1.ConditionFalse,
-			Reason:  "BackupNotDone",
-			Message: "Backup is required and not done"})
+// setBackupStartedCondition sets conditions of backup required
+func (r *ClusterGroupUpgradeReconciler) setBackupStartedCondition(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
+	utils.SetStatusCondition(
+		&clusterGroupUpgrade.Status.Conditions,
+		utils.ConditionTypes.BackupSuceeded,
+		utils.ConditionReasons.InProgress,
+		metav1.ConditionFalse,
+		"Backup is required and not done",
+	)
 }
 
 // checkAllBackupDone handles alleviation of BackupDone==False condition
 func (r *ClusterGroupUpgradeReconciler) checkAllBackupDone(
 	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
 	// Handle completion
-	if func() bool {
-		for _, state := range clusterGroupUpgrade.Status.Backup.Status {
-			if state != BackupStateSucceeded {
-				return false
-			}
+
+	var failedBackupClusters []string
+	var progressingBackupClusters []string
+
+	for clusterName, state := range clusterGroupUpgrade.Status.Backup.Status {
+		if state == BackupStateActive || state == BackupStateStarting || state == BackupStatePreparingToStart {
+			progressingBackupClusters = append(progressingBackupClusters, clusterName)
+		} else if state == BackupStateError || state == BackupStateTimeout {
+			failedBackupClusters = append(failedBackupClusters, clusterName)
+		}
+	}
+
+	if len(progressingBackupClusters) == 0 && len(failedBackupClusters) == 0 {
+		utils.SetStatusCondition(
+			&clusterGroupUpgrade.Status.Conditions,
+			utils.ConditionTypes.BackupSuceeded,
+			utils.ConditionReasons.Completed,
+			metav1.ConditionTrue,
+			"Backup is completed for all clusters",
+		)
+	} else if len(progressingBackupClusters) == 0 && len(failedBackupClusters) > 0 {
+		if len(failedBackupClusters) == len(clusterGroupUpgrade.Status.Backup.Status) {
+			utils.SetStatusCondition(
+				&clusterGroupUpgrade.Status.Conditions,
+				utils.ConditionTypes.BackupSuceeded,
+				utils.ConditionReasons.Failed,
+				metav1.ConditionTrue,
+				"Backup failed for all clusters",
+			)
+		} else {
+			utils.SetStatusCondition(
+				&clusterGroupUpgrade.Status.Conditions,
+				utils.ConditionTypes.BackupSuceeded,
+				utils.ConditionReasons.PartiallyDone,
+				metav1.ConditionTrue,
+				fmt.Sprintf("Backup failed for clusters: [%s]", strings.Join(failedBackupClusters, ",")),
```

Comment:
For me, the message for which cluster failed is not necessary, suppose, if we launch the CU having 100 clusters, and in the worst case, all the cluster failed, we are not going to overload the message variable with tons of strings.